crypto/rand/rand_unix.c

```diff
@@ -149,26 +163,94 @@ size_t rand_pool_acquire_entropy(RAND_POOL *pool)
 #   error "Seeding uses urandom but DEVRANDOM is not configured"
 #  endif
 
+#  if defined(__GLIBC__) && defined(__GLIBC_PREREQ)
+#   if __GLIBC_PREREQ(2, 25)
+#    define OPENSSL_HAVE_GETRANDOM
+#   endif
+#  endif
+
+#  if (defined(__FreeBSD__) && __FreeBSD_version >= 1200061)
```

Comment:
Worth splitting this like was done for `__GLIBC_PREREQ` ?

If neither `__FreeBSD__` nor `__FreeBSD_version` is defined, isn't there is a syntax error?

Reply:
The reason for the __GLIBC_PREREQ() on the next line is that it takes parameters, so you need to be sure it exist before you use it.